### PR TITLE
Bluntsord/resource implementation

### PR DESCRIPTION
**Pull request by Bluntsord**
made enemies drop gold. Currently gold is displayed when the enemies are die. But due to the modulating of the entire enemy, the text gets colored too. Think we can change it? im okay with both. Also, theres a bug where sometimes enemy dies twice. Its rare but when there are alot of towers shooting one enemy. Again, think it might not make so much diff unless its a boss or smth lol.

### CODE REVIEW

**Comment by bharathcs** [comment]
Nice looks good to me. I'm fine with the text being colored or not lol. You can fix james bond bug by moving the new lines of code to the end of the function:
image

**Review by bharathcs — COMMENTED** [review]
lgtm!

**Comment by Bluntsord** [comment]
works

File: game/src/environment/towers/core/Tower.gd [hunk 1]
```diff
@@ -40,6 +43,7 @@ func _ready() -> void:
 	_reset()
 	# warning-ignore:return_value_discarded
 	_timer.connect("timeout", self, "_fire")
+	$Range/RangeRadius.shape.radius = range_radius
```
Comment (bharathcs) [inline game/src/environment/towers/core/Tower.gd:46]: Shouldn't this be moved into` _reset()`

File: game/src/ui/gameplay/Arena/tile-selector/TileSelector.gd [hunk 2]
```diff
@@ -79,11 +79,23 @@ func take_action():
 	var tower_inventory = arena.get_node("UI/Control/TowerInventory")
 	match _action:
 		ACTION.BUILDING:
-			arena.build_tower(_x, _y, _type)
 			var mouse_tower = self.get_node("Tower")
+			var tower_cost = mouse_tower.gold_cost
+			var negative_tower_cost = tower_cost * -1
```
Comment (bharathcs) [inline game/src/ui/gameplay/Arena/tile-selector/TileSelector.gd:84]: Is there a reason u want to do this here?

I think its bad UI to have the unaffordable towers clickable, but not buildable.

File: game/src/ui/gameplay/BuildTool/GoldCost.gd [hunk 3]
```diff
@@ -0,0 +1,16 @@
+extends Node2D
```
Comment (bharathcs) [inline game/src/ui/gameplay/BuildTool/GoldCost.gd]: Whats this file here for, can we delete? Its not even referenced in GoldCost.tscn

File: game/src/ui/gameplay/BuildTool/GoldLabel.gd [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+func can_drop_data(position, data):
+	return true
+
+func drop_data(position, data):
+	data["target_slot"] = data["origin_slot"]
+	data["target_item"] = data["origin_item"]
+	data["origin_slot"].texture = data["origin_texture"]
+
+func reset():
+	_ready()
```
Comment (bharathcs) [inline game/src/ui/gameplay/BuildTool/GoldLabel.gd]: Oof you should never call `_ready` like this. Much better to do the following:
```
func _ready():
	reset()

func reset():
	curr_gold = start_gold
	update_gold()
```

File: game/src/ui/gameplay/UI/UI.tscn [hunk 5]
```diff
@@ -13,12 +13,16 @@
...
 [ext_resource path="res://assets/img/UI/BuildToolUIBackground.png" type="Texture" id=12]
 [ext_resource path="res://assets/img/towers/pixelTowers/Demon_Statue/Demon_Statue_red_sheet.png" type="Texture" id=13]
+[ext_resource path="res://assets/img/items/Pixel Art Icon Pack - RPG/Ore & Gem/Gold Nugget.png" type="Texture" id=14]
 [ext_resource path="res://assets/img/towers/pixelTowers/LightningTotemFull/Totem_full-Sheet(Red).png" type="Texture" id=15]
 [ext_resource path="res://assets/img/towers/pixelTowers/BloodMoonTower_full_version/RedMoonTower_free_idle_animation..png" type="Texture" id=16]
+[ext_resource path="res://src/ui/gameplay/BuildTool/GoldLabel.gd" type="Script" id=17]
 [ext_resource path="res://src/ui/gameplay/BuildTool/tower-builds.gd" type="Script" id=18]
 [ext_resource path="res://src/ui/gameplay/UI/UI.gd" type="Script" id=19]
 [ext_resource path="res://assets/img/UI/Cartoon GUI/PNG/Item Slot/Cartoon RPG UI_Slot - Grade S.png" type="Texture" id=20]
+[ext_resource path="res://src/ui/gameplay/GoldIndicator/GoldIndicator.gd" type="Script" id=21]
```
Comment (bharathcs) [inline game/src/ui/gameplay/UI/UI.tscn:23]: I don't see the GoldIndicator.tscn file anywhere in this PR

File: game/src/ui/gameplay/towerInventory/TowerInventory.gd [hunk 6]
```diff
@@ -70,7 +70,10 @@ func change_tower_to_be_build(tower: Tower):
 		return
 	tower_to_be_built = tower
 	tower_inventory_items_held = tower_to_be_built.tower_items_held
-	tower_inventory_items_held = tower_inventory_items_held.duplicate(true)
```
Comment (bharathcs) [inline game/src/ui/gameplay/towerInventory/TowerInventory.gd]: Please `git merge main` to get the latest fixes for tower inventory.gd

**Review by bharathcs — CHANGES_REQUESTED** [review]
Hey my dude, I like the resource management, but i do have some uestions about the implementation.

**Comment by bharathcs** [comment]
My apologies one more thing:

this shouldn't be here, (I must have approved this in some PR and missed it sorry about that), could you move this to the right place:
image

File: game/src/environment/towers/core/Tower.gd (hunk 1 above)
Comment (Bluntsord) [reply game/src/environment/towers/core/Tower.gd:46]: either way its wrong. Think i have to relook at the tower range things.

File: game/src/ui/gameplay/Arena/tile-selector/TileSelector.gd (hunk 2 above)
Comment (Bluntsord) [reply game/src/ui/gameplay/Arena/tile-selector/TileSelector.gd:84]: i thought it was better no? incase they wanna see the range of the towers etc

File: game/src/ui/gameplay/BuildTool/GoldCost.gd (hunk 3 above)
Comment (Bluntsord) [reply game/src/ui/gameplay/BuildTool/GoldCost.gd]: removed

File: game/src/ui/gameplay/BuildTool/GoldLabel.gd (hunk 4 above)
Comment (Bluntsord) [reply game/src/ui/gameplay/BuildTool/GoldLabel.gd]: changed

File: game/src/ui/gameplay/UI/UI.tscn (hunk 5 above)
Comment (Bluntsord) [reply game/src/ui/gameplay/UI/UI.tscn:23]: created it just in the UI scene. Doesnt have its own scene just .gd

File: game/src/environment/towers/core/Tower.gd (hunk 1 above)
Comment (bharathcs) [reply game/src/environment/towers/core/Tower.gd:46]: TC said handle later (on tele): https://github.com/quantum-defence/quantum-defence/issues/89

File: game/src/ui/gameplay/BuildTool/GoldLabel.gd (hunk 4 above)
Comment (bharathcs) [reply game/src/ui/gameplay/BuildTool/GoldLabel.gd]: Settled.

**Review by Bluntsord — COMMENTED** [review]
solved. Moved get chest to an appropriate folder

File: game/src/ui/gameplay/Arena/tile-selector/TileSelector.gd (hunk 2 above)
Comment (bharathcs) [reply game/src/ui/gameplay/Arena/tile-selector/TileSelector.gd:84]: https://github.com/quantum-defence/quantum-defence/issues/90

File: game/src/ui/gameplay/towerInventory/TowerInventory.gd (hunk 6 above)
Comment (bharathcs) [reply game/src/ui/gameplay/towerInventory/TowerInventory.gd]: Not merged.

Comment (bharathcs) [reply game/src/ui/gameplay/towerInventory/TowerInventory.gd]: Done with latest commit 74b330a6975bf1f171f8d393a0884749b0246fc6

**Pull request merged**